### PR TITLE
CBAPI-2380 - CBC SDK - add recommendation code

### PR DESCRIPTION
**Pull request by abowersox-cb**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Tests have been added that prove the fix is effective or that the feature works.
- [x] New and existing tests pass locally with the changes.
- [x] Code follows the style guidelines of this project (PEP8, clean code, linter).
- [x] A self-review of the code has been done.

## What is the ticket or issue number?

https://jira.carbonblack.local/browse/CBAPI-2380

## Pull Request Description

Add support for recommendation API to CBC SDK.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## How Has This Been Tested?

Unit test coverage for newly added source is at 95%.

### CODE REVIEW

File: src/cbc_sdk/endpoint_standard/recommendation.py [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+    def __init__(self, cb, model_unique_id, initial_data=None):
+        """
+        Initialize the Recommendation object.
+
+        Args:
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+            model_unique_id (str): ID of the recommendation represented.
+            initial_data (dict): Initial data used to populate the recommendation.
+        """
+        super(Recommendation, self).__init__(cb, model_unique_id, initial_data, full_doc=True)
```
Comment (emitreva-cb) [inline src/cbc_sdk/endpoint_standard/recommendation.py:45]: If I saw correctly there is no /recommendation-service/v1/orgs/{org_key}/recommendation/{id}, but just search, so maybe we do not need model_unique_id. If you do not think that's correct just disregard.

Comment (abowersox-cb) [reply src/cbc_sdk/endpoint_standard/recommendation.py:45]: It gets initialized when those objects get created in the query, with the `recommendation_id` member of the item data. And I _do_ need it when putting together the URL in the `_take_action` function.

File: src/cbc_sdk/endpoint_standard/recommendation.py [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+        """
+        Reset the recommendation, undoing any created reputation override and setting it back to NEW state.
+
+        Args:
+            comment (str): Optional comment associated with the action.
+        """
+        self._take_action('RESET', comment)
+
+
+class RecommendationImpact(UnrefreshableModel):
```
Comment (avanbrunt-cb) [inline src/cbc_sdk/endpoint_standard/recommendation.py:136]: Why make this a separate class instead of nesting inside the recommendation class?

Comment (avanbrunt-cb) [reply src/cbc_sdk/endpoint_standard/recommendation.py:136]: Same for the other nested classes too

Comment (abowersox-cb) [reply src/cbc_sdk/endpoint_standard/recommendation.py]: OK, I did that.

**Review by avanbrunt-cb — APPROVED** [review]
LGTM

**Pull request merged**